src/components/pages/[...params].js

```diff
@@ -4,11 +4,50 @@ import { useRouter } from 'next/router';
 import { Box } from '@/ui/Box';
 import { useCookiesWithOptions } from '@/hooks/useCookiesWithOptions';
 import { getApplicationServerSideProps } from '@/utils/getApplicationServerSideProps';
-import { memo, useMemo } from 'react';
+import { memo, useMemo, useState } from 'react';
+import { generatePath, matchPath } from 'react-router';
+import { getRedirects } from '../../redirects';
+import {
+  useHandleWindowMessage,
+  WindowMessageTypes,
+} from '@/hooks/useHandleWindowMessage';
+import PageNotFound from '@/pages/404';
+import { useIsClient } from '@/hooks/useIsClient';
+import { isFeatureFlagEnabledInCookies } from '@/hooks/useFeatureFlag';
+import { Cookies } from 'react-cookie';
 
 const prefixWhenNotEmpty = (value, prefix) =>
   value ? `${prefix}${value}` : value;
 
+const getRedirect = (originalPath, environment, cookies) => {
+  const matches = getRedirects(environment)
+    .map(({ source, destination, permanent, featureFlag }) => {
+      // Don't follow redirects that are behind a feature flag
+      if (featureFlag) {
+        const enabled = isFeatureFlagEnabledInCookies(featureFlag, cookies);
+        if (!enabled) {
+          return false;
+        }
+      }
+
+      // Check if the redirect source matches the current path
+      const match = matchPath(originalPath, { path: source, exact: true });
+      if (match) {
+        return {
+          destination: generatePath(destination, match.params),
+          permanent: featureFlag === undefined && permanent,
+        };
+      }
+      return false;
+    })
+    .filter((match) => !!match);
```

Comment:
We could also use a find here

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/find

Reply:
Done! I wasn't able to remove the `map` though because `find` only filters the array, it doesn't allow us to change the values or return a new value.